webauthn/test/integration/SafeWebAuthn.test.ts

```diff
@@ -0,0 +1,207 @@
+import hre from "hardhat";
+import { expect } from "chai";
+import { AddressZero } from "@ethersproject/constants";
+import { getBytes, concat, resolveProperties, ethers } from "ethers";
+import { ethers as ethersV5 } from "ethers-v5";
+import { UserOperationStruct } from "@account-abstraction/contracts";
+import { getUserOpHash } from "@account-abstraction/utils";
+import { calculateProxyAddress } from "../../utils/calculateProxyAddress";
+
+const ERC4337_TEST_ENV_VARIABLES_DEFINED =
+  typeof process.env.ERC4337_TEST_BUNDLER_URL !== "undefined" &&
+  typeof process.env.ERC4337_TEST_NODE_URL !== "undefined" &&
+  typeof process.env.ERC4337_TEST_SAFE_FACTORY_ADDRESS !== "undefined" &&
+  typeof process.env.ERC4337_TEST_SINGLETON_ADDRESS !== "undefined" &&
+  typeof process.env.MNEMONIC !== "undefined";
+
+const itif = ERC4337_TEST_ENV_VARIABLES_DEFINED ? it : it.skip;
+const SAFE_FACTORY_ADDRESS = process.env.ERC4337_TEST_SAFE_FACTORY_ADDRESS;
+const SINGLETON_ADDRESS = process.env.ERC4337_TEST_SINGLETON_ADDRESS;
+const BUNDLER_URL = process.env.ERC4337_TEST_BUNDLER_URL;
+const NODE_URL = process.env.ERC4337_TEST_NODE_URL;
+const MNEMONIC = process.env.MNEMONIC;
+
+const sleep = (ms: number) => new Promise((resolve) => setTimeout(resolve, ms));
+
+describe("Safe.ERC4337", () => {
+  const setupTests = async () => {
+    const factory = await hre.ethers.getContractFactory("SafeProxyFactory");
+    const singleton = await hre.ethers.getContractFactory("Safe");
+
+    const bundlerProvider = new ethersV5.providers.JsonRpcProvider(BUNDLER_URL);
```

Comment:
The `bundlerProvider` did not play nicely with ethers v6. So installed ethers v5 and aliased it (see ethers-v5 in package.json).

Example of the errors I was getting with an ethers v6 provider attached to the bundler:
```
JsonRpcProvider failed to detect network and cannot start up; retry in 1s (perhaps the URL is wrong or the node is not started)
```
```
    Error: missing response for request (value=[ { "error": { "code": -32601,
    "message": "Method undefined is not supported" } } ], info={ "payload": { "id": 2,
    "jsonrpc": "2.0", "method": "eth_supportedEntryPoints", "params": [  ] } },
    code=BAD_DATA, version=6.6.7)
```